homeassistant/components/switchbot_cloud/vacuum.py

```diff
@@ -90,30 +103,149 @@ async def async_set_fan_speed(self, fan_speed: str, **kwargs: Any) -> None:
         await self.send_api_command(VacuumCommands.DOCK)
+        self.async_write_ha_state()
 
     async def async_start(self) -> None:
         """Start or resume the cleaning task."""
         await self.send_api_command(VacuumCommands.START)
+        self.async_write_ha_state()
 
     def _set_attributes(self) -> None:
         """Set attributes from coordinator data."""
-        if not self.coordinator.data:
+        if self.coordinator.data is None:
             return
 
         self._attr_battery_level = self.coordinator.data.get("battery")
         self._attr_available = self.coordinator.data.get("onlineStatus") == "online"
 
         switchbot_state = str(self.coordinator.data.get("workingStatus"))
         self._attr_activity = VACUUM_SWITCHBOT_STATE_TO_HA_STATE.get(switchbot_state)
+        if self._attr_fan_speed is None:
+            self._attr_fan_speed = VACUUM_FAN_SPEED_QUIET
+
+
+class SwitchBotCloudVacuumK20PlusPro(SwitchBotCloudVacuum):
+    """Representation of a SwitchBot K20+ Pro."""
+
+    async def async_set_fan_speed(self, fan_speed: str, **kwargs: Any) -> None:
+        """Set fan speed."""
+        self._attr_fan_speed = fan_speed
+        await self.send_api_command(
+            VacuumCleanerV2Commands.CHANGE_PARAM,
+            parameters={
+                "fanLevel": int(VACUUM_FAN_SPEED_TO_SWITCHBOT_FAN_SPEED[fan_speed]) + 1,
+                "waterLevel": 1,
+                "times": 1,
+            },
+        )
+        self.async_write_ha_state()
+
+    async def async_pause(self) -> None:
+        """Pause the cleaning task."""
+        await self.send_api_command(VacuumCleanerV2Commands.PAUSE)
+        self.async_write_ha_state()
+
+    async def async_return_to_base(self, **kwargs: Any) -> None:
+        """Set the vacuum cleaner to return to the dock."""
+        await self.send_api_command(VacuumCleanerV2Commands.DOCK)
+        self.async_write_ha_state()
+
+    async def async_start(self) -> None:
+        """Start or resume the cleaning task."""
+        fan_level = (
+            VACUUM_FAN_SPEED_TO_SWITCHBOT_FAN_SPEED.get(self.fan_speed)
+            if self.fan_speed
+            else None
+        )
+        await self.send_api_command(
+            VacuumCleanerV2Commands.START_CLEAN,
+            parameters={
+                "action": VacuumCleanMode.SWEEP.value,
+                "param": {
+                    "fanLevel": int(fan_level if fan_level else VACUUM_FAN_SPEED_QUIET)
+                    + 1,
+                    "times": 1,
+                },
+            },
+        )
+        self.async_write_ha_state()
+
+
+class SwitchBotCloudVacuumK10PlusProCombo(SwitchBotCloudVacuumK20PlusPro):
+    """Representation of a SwitchBot vacuum K10+ Pro Combo."""
+
+    async def async_set_fan_speed(self, fan_speed: str, **kwargs: Any) -> None:
+        """Set fan speed."""
+        self._attr_fan_speed = fan_speed
+        if fan_speed in VACUUM_FAN_SPEED_TO_SWITCHBOT_FAN_SPEED:
```

Comment:
why do we extend this only for the K10?

Reply:
The only difference is the parameters
k20 pro
<img width="732" height="274" alt="image" src="https://github.com/user-attachments/assets/ad98edcf-61e3-4c1e-8034-78acc1f94783" />
K10+ Pro Combo
<img width="758" height="288" alt="image" src="https://github.com/user-attachments/assets/a17be355-0106-4063-b9e5-7494f496f1b6" />